ios/RNSScreenStackAnimator.mm

```diff
@@ -128,22 +132,33 @@ - (void)animateSimplePushWithShadowEnabled:(BOOL)shadowEnabled
       shadowView.alpha = 0.0;
     }
 
-    [UIView animateWithDuration:[self transitionDuration:transitionContext]
-        animations:^{
-          fromViewController.view.transform = leftTransform;
-          toViewController.view.transform = CGAffineTransformIdentity;
-          if (shadowView) {
-            shadowView.alpha = RNSShadowViewMaxAlpha;
-          }
-        }
-        completion:^(BOOL finished) {
-          if (shadowView) {
-            [shadowView removeFromSuperview];
-          }
-          fromViewController.view.transform = CGAffineTransformIdentity;
-          toViewController.view.transform = CGAffineTransformIdentity;
-          [transitionContext completeTransition:![transitionContext transitionWasCancelled]];
-        }];
+    // Default curve provider is as defined below, however defined this way it ignores the `animationDuration` prop.
+    // Damping of 1.0 seems close enough.
+    // id<UITimingCurveProvider> timingCurveProvider = [[UISpringTimingParameters alloc] init];
+
+    id<UITimingCurveProvider> timingCurveProvider = [[UISpringTimingParameters alloc] initWithDampingRatio:1.0];
```

Comment:
I think such config is definitely better, but animation begins very slowly (unlike native iOS animation). Compare it yourself (I took 3rd frame for a reference):

<img width="456" alt="image" src="https://github.com/user-attachments/assets/6ffdce5e-4133-40e3-8903-2fc63fe2d895">

I think next config gives a better results:

```objc
id<UITimingCurveProvider> timingCurveProvider = [[UISpringTimingParameters alloc] initWithMass:3 stiffness:1000 damping:500 initialVelocity:CGVectorMake(0, 0)];
```

<img width="453" alt="image" src="https://github.com/user-attachments/assets/4ac802dd-a43b-4710-99b7-2a5a64a54617">

Let me see if such config supports custom duration 👀

Reply:
Okay, even though a new config is matching a native animation more closely - it doesn't allow to customize duration in this case 😔

Reply:
Most likely it does not, but let me know.

The spring curve for default animation is just `[[UISpringTimingParameters alloc] init]` (according to "iOS Programming Guide" by Matt Neuburg, there is no need to guess the params or use some magic constants 😅 )

Reply:
I've opened https://github.com/software-mansion/react-native-screens/pull/2477, as making this work requires more changes that I've thought initially. I'm in process of writing PR description but have some meetings currently.

Reply:
@kkafar but if we are using `damping: 1`, then we have following transition:

|Variant         |1 frame|2 frame|3 frame|4 frame|5 frame|
|------------|-------|--------|--------|--------|--------|
|damping: 1 |<img width="454" alt="image" src="https://github.com/user-attachments/assets/4db0bbc6-dffe-4bd9-a8a2-742cafa42a0d">|<img width="452" alt="image" src="https://github.com/user-attachments/assets/3efd44ef-5d6d-42d1-b3f7-c2478c59a63a">|<img width="451" alt="image" src="https://github.com/user-attachments/assets/4c7ef215-e967-45cc-9e42-ee0bf904e3a8">|<img width="450" alt="image" src="https://github.com/user-attachments/assets/be6354a4-b696-4069-a279-033261ddf4e5">|<img width="450" alt="image" src="https://github.com/user-attachments/assets/6d48bf72-a8e7-4ce3-9938-aa19f82c1e0a">|
|Native iOS  |<img width="447" alt="image" src="https://github.com/user-attachments/assets/b8c4895f-6d32-4744-8283-25af0c59d881">|<img width="449" alt="image" src="https://github.com/user-attachments/assets/197386af-f659-4f0d-9749-e07757246951">|<img width="446" alt="image" src="https://github.com/user-attachments/assets/34e9c0eb-b1dc-484e-a5e9-271faf207ace">|<img width="448" alt="image" src="https://github.com/user-attachments/assets/590d4042-9679-4d98-a5ad-a0419f77bacc">|<img width="447" alt="image" src="https://github.com/user-attachments/assets/f3541c1c-7f6d-4baf-9b57-74afa74920a8">|

For me it looks like a native iOS animation (`default`) is smoother than `simple_push` (with `damping: 1` it feels like it's moving slightly faster than I expect to).